src/Snippet.tsx

```diff
@@ -1,28 +1,32 @@
-import * as React from 'react';
+import React from 'react';
 
-import './snippet.css';
+import Prism from 'prismjs';
+import 'prismjs/components/prism-python.min.js';
+import 'prismjs/themes/prism-okaidia.css';
 
 import * as Block from './Block'
+import './snippet.css';
+
 
 interface ISnippetProps {
   blocks: Block.IBlock[];
   lang: string;
 }
 
 class Snippet extends React.Component<ISnippetProps, {}> {
-  constructor(props: ISnippetProps) {
-    super(props);
-    this.state = {};
+  // Note that routing doesn't necessarily remount components, so we need to
+  // re-highlight on update instead of mounting to make sure we always have
+  // highlighted code.
+  public componentDidUpdate() {
+    Prism.highlightAll();
```

Comment:
Are we sure we can't just do this with a element ref and Prism.highlightElement? Anyway looks fine for now

Reply:
Maybe refs could work, but the way we did it previously didn't. I'm moving ahead to more layout stuff, so I won't be looking into it.